### DISCUSSION TITLE
404s within selected zone

### DISCUSSION THREAD

**Discussion by helloimalastair** [protomaps/PMTiles#119] [Q&A] [answered]
Hey y'all!

Came across this project yesterday, and tried setting an instance up with Workers/R2. Weirdly though, the range requests on R2 seem to be returning nothing.

Map file is based on the [Geofabrik Denmark](https://download.geofabrik.de/europe/denmark.html) download, and even with a view set directly within the capital region, it still returns a `404 Archive Not Found` [Example](https://maps.goalastair.com/tiles/8727.5162.14.mvt)

Anyone have any idea what I might be missing?

**Edit**: [Script](https://gist.github.com/helloimalastair/09deaab1aef8e9137e7c57cd33b97b63) for reference

**Edit 2**: I thought maybe it had something to do with the fact that I tried to rewrite the CF Workers driver, but it still seems to show the same error [here](https://dev-maps.goalastair.com/tiles/8727/5162/14.mvt).

**Comment by bdon** [comment, marked as answer]
Your URL is `https://dev-maps.goalastair.com/tiles/8727/5162/14.mvt`

Is there a file called `tiles.pmtiles` in the root of your bucket?

**Reply by helloimalastair** [reply]
Oops, my mistake. The modified version I created completely ignores the `/{filename}` segment of the URL, so I forgot to rewrite the dev version of the URL I posted. The file is called `map.pmtiles` in my bucket, which means the URL is [https://dev-maps.goalastair.com/map/8727/5162/14.mvt](https://dev-maps.goalastair.com/map/8727/5162/14.mvt), which also doesn't seem to load.

**Reply by bdon** [reply]
If you can deploy https://protomaps.com/docs/cdn/cloudflare successfully, then the issue lies with your modifications to the worker. It's difficult to see what you modified based on the bundled script above, and it would be easier if you could show the source code as well as explain what modifications you made.

**Reply by helloimalastair** [reply]
That's the thing. I originally only deployed my modified setup, and noticed a 404([https://maps.goalastair.com/tiles/8727.5162.14.mvt](https://maps.goalastair.com/tiles/8727.5162.14.mvt)).

To cover my bases, I also deployed the Worker directly from the tutorial, as explained, as closely as possible, which is the URL I provided from above in my reply([https://dev-maps.goalastair.com/map/8727/5162/14.mvt](https://dev-maps.goalastair.com/map/8727/5162/14.mvt)).

For clarification, this is my entire `wrangler.toml` for the default setup, modified as minimally as possible:
```toml
name = "pmtiles-cloudflare"
main = "src/index.ts"
compatibility_date = "2022-10-18"
minify = true
route = { pattern = "dev-maps.goalastair.com", custom_domain = true }

r2_buckets  = [
  { binding = "BUCKET", bucket_name = "protomap" }
]

[vars]
ALLOWED_ORIGINS = "*"
```

As far as I can tell, neither work as expected, so I'm not 100% sure what else the issue might be...

**Reply by bdon** [reply]
also FWIW tiles are in Z/X/Y order, not X/Y/Z, so the URLs should look like https://dev-maps.goalastair.com/tiles/14/8727/5162.mvt

The HTTP response I get from your worker is "Archive not found", so I would ensure that the worker has read access to the bucket and the `tiles.pmtiles` exists there. But it's difficult to debug further without access to your Cloudflare account.

**Reply by bdon** [reply]
I would test with the tile`0/0/0.mvt` assuming your tile generation tool creates complete pyramids up to z0

**Reply by helloimalastair** [reply]
Ok, so this is kind of embarassing. Apparently, when I first uploaded the file to the bucket, I uploaded `denmark.mbtiles` to a folder called `map.pmtiles`... 🤦

I fixed that now, and ran a global cache purge, but it still doesn't seem to like `https://dev-maps.goalastair.com/tiles/14/8727/5162.mvt` nor `https://dev-maps.goalastair.com/tiles/0/0/0.mvt`.

As for your question about support `z0` zoom levels, I'm not actually 100% sure what my dataset supports. If it isn't obvious, this is my first real foray into OSM/Map Tile Generation/Serving, other than having used Mapbox/a Docker Geocoder for a little while. I kind of bodged a pipeline together to get a `pmtiles` file.

Basically, I download an `.osm.pbf` file from Geofabrik, run it through `tilemaker` to convert to `mbtiles`, and then through the `pmtiles` CLI. In the end, I am not sure what zoom levels it produces, since I didn't have any input there. Though, taking a guess from [this line](https://github.com/systemed/tilemaker/blob/57942b9812e1e3c7b5ec0f46820fa853f36f3c44/resources/config-openmaptiles.json#L37) in the Tilemaker Config, I would assume it would support `z0` zoom?

**Reply by bdon** [reply]
You can drag and drop your `.pmtiles` file to https://protomaps.github.io/PMTiles/ and look at metadata > min zoom to determine what zoom levels are supported. If you want to screenshot what the root of your bucket looks like in the CF web UI that would be valuable information.

**Reply by helloimalastair** [reply]
Here's a copy of the metadata, which shows the min zoom is, or at least should be, `z0`:
```
root directory: offset=127 len=89
metadata: offset=216 len=511
leaf directories: offset=727 len=108182
tile data: offset=108909 len=258121105
num addressed tiles: 72904
num tile entries: 48720
num tile contents: 45678
clustered: true
internal compression: 2
tile compression: 2
tile type: 1
min zoom: 0
max zoom: 14
min lon, min lat, max lon, max lat: 7.7011, 54.44065, 15.65449, 58.06239
center zoom: 7
center lon, center lat: 11.677795, 56.25152
```
And here's a screenshot of my entire R2 bucket. It is just the `map.pmtiles` file, and nothing else.
![Screenshot of Protomap Bucket](https://user-images.githubusercontent.com/37487746/218732963-cb738124-458b-4e14-be6a-465569a14323.png)

**Reply by helloimalastair** [reply]
Wait... Not again...

Clearly I have no idea what I'm doing...

[https://maps.goalastair.com/tiles/0.0.0.mvt](https://maps.goalastair.com/tiles/0.0.0.mvt)
[https://dev-maps.goalastair.com/tiles/0/0/0.mvt](https://maps.goalastair.com/tiles/0.0.0.mvt)

Both work now...

**Reply by helloimalastair** [reply]
Ok, so final question. Does the map style have any effect on what is rendered? I'm running base `protomap.js`, and it is only rendering an outline of the landmass, and no roads/POIs/anything else.

Viewable on a [Repl](https://meanunusedcensorware.helloimalastair.repl.co/#8/57.283/11.706)

**Reply by bdon** [reply]
Hi, If you are using Tilemaker it uses the OpenMapTiles schema definition as linked in that tilemaker config file, while by default protomaps.js will use layers and tags as defined in these docs: https://protomaps.com/docs/frontends/basemap-layers

You can find JSON styles for using with MapLibre in the OpenMapTiles org.

**Reply by helloimalastair** [reply]
I'm still not 100% sure I follow. Do I need to use the styles on the Tilemaker Config? Or do I need to apply it to `protomap.js` somehow?

**Reply by bdon** [reply]
If your plan is to use OpenMapTiles you should use MapLibre GL JS, not leaflet or protomaps.js.

**Reply by helloimalastair** [reply]
Ok, thank you for all your help!

**Reply by helloimalastair** [reply]
~~Hey, sorry for *necro*ing this post, but just wanted to check, is there something I'm still missing? Running [helloimalastair/protomaps](https://github.com/helloimalastair/protomaps), and preview available at [https://dev.maps.goalastair.com](https://dev.maps.goalastair.com). I am able to get it working with the Protomaps provider for MapLibre([preview](https://maps.goalastair.com)), but when I try to implement the Worker-Side Range System, it doesn't even seem to make any requests...~~
Never mind, it was just pulling from an old config... 🤦
